src/server/test/db/barchartReadingsTests.js

```diff
@@ -26,18 +16,23 @@ mocha.describe('Barchart Readings', () => {
 	const timestamp5 = timestamp4.clone().add(1, 'hour');
 
 	mocha.beforeEach(async () => {
-		await new Meter(undefined, 'Meter', null, false, Meter.type.MAMAC).insert();
-		meter = await Meter.getByName('Meter');
+		conn = testDB.getConnection();
+		await new Meter(undefined, 'Meter', null, false, Meter.type.MAMAC).insert(conn);
+		meter = await Meter.getByName('Meter', conn);
 	});
 
 	mocha.it('barchart readings with no specified time interval', async () => {
+		conn = testDB.getConnection();
 		const reading1 = new Reading(meter.id, 100, timestamp1, timestamp2);
 		const reading2 = new Reading(meter.id, 200, timestamp2, timestamp3);
 		const reading3 = new Reading(meter.id, 300, timestamp3, timestamp4);
 		const reading4 = new Reading(meter.id, 400, timestamp4, timestamp5);
-		await Reading.insertAll([reading1, reading2, reading3, reading4]);
+		await Reading.insertAll([reading1, reading2, reading3, reading4], conn);
 
-		const barchartReadings = await Reading.getBarchartReadings([meter.id], moment.duration(2, 'h'));
+		// TODO: This is awkward. Is there a better way to have conn as a non-optional
+		// without requiring changing EVERY call to getBarchartReadings?
+		// If not, well, we need to change every call.
+		const barchartReadings = await Reading.getBarchartReadings([meter.id], moment.duration(2, 'h'), null, null, conn);
```

Comment:
I think it is just fine to change all of these calls because then it is easy to see where exactly `conn` is used.

Reply:
What I meant here was to eliminate the `null, null` near the end.

Reply:
Sounds good. I can tell you I have absolutely zero clue how to do that.